Extensions/xkit_patches.js

```diff
@@ -269,6 +269,329 @@ XKit.extensions.xkit_patches = new Object({
+				}
+
+				if (XKit.interface.where().search) {
+					XKit.interface.trigger_reflow();
+				}
+			};
+
+			/**
+			 * @param {String} post_id
+			 * @return {Object} Interface Post Object of post with given id
+			 */
+			XKit.interface.find_post = function(post_id) {
+
+				// Return a post object based on post ID.
+
+				if ($("body").find("#post_" + post_id).length > 0) {
+					return XKit.interface.parse_post($("#post_" + post_id));
+				} else if ($(".mh_post").length > 0) {
+					return XKit.interface.parse_post($(".mh_post"));
+				} else {
+					var m_error = {};
+					m_error.error = true;
+					m_error.error_message = "Object not found on page.";
+					return m_error;
+				}
+
+			};
+
+			XKit.interface.post = { /* globals Promise */
+
+				fetch: function(post) {
+					return new Promise(function(resolve, reject) {
+						XKit.tools.Nx_XHR({
+							method: "POST",
+							url: "https://www.tumblr.com/svc/post/fetch",
+							headers: {
+								"X-Tumblr-Form-Key": XKit.interface.form_key(),
+								"X-Requested-With": "XMLHttpRequest"
+							},
+							json: true, data: JSON.stringify(post),
+							onerror: function(response) {
+								reject(response);
+							},
+							onload: function(response) {
+								try {
+									resolve(JSON.parse(response.responseText));
+								} catch (e) {
+									reject(response);
+								}
+							}
+						});
+					});
+				},
+
+				update: function(post) {
+					return new Promise(function(resolve, reject) {
+						XKit.interface.kitty.get(function(kitty) {
+							if (kitty.errors) {
+								reject(kitty.response);
+							}
+
+							XKit.tools.Nx_XHR({
+								method: "POST",
+								url: "https://www.tumblr.com/svc/post/update",
+								headers: {
+									"X-Tumblr-Puppies": kitty.kitten,
+									"X-Tumblr-Form-Key": XKit.interface.form_key(),
+									"X-Requested-With": "XMLHttpRequest"
+								},
+								json: true, data: JSON.stringify(post),
+								onerror: function(response) {
+									reject(response);
+								},
+								onload: function(response) {
+									try {
+										resolve(JSON.parse(response.responseText));
+									} catch (e) {
+										reject(response);
+									}
+								}
+							});
+						});
+					});
+				},
+
+				delete: function(post) {
+					return new Promise(function(resolve, reject) {
+						XKit.tools.Nx_XHR({
+							method: "POST",
+							url: "https://www.tumblr.com/svc/post/delete",
+							headers: {
+								"X-Tumblr-Form-Key": XKit.interface.form_key(),
+								"X-Requested-With": "XMLHttpRequest"
+							},
+							json: true, data: JSON.stringify(post),
+							onerror: function(response) {
+								reject(response);
+							},
+							onload: function(response) {
+								try {
+									resolve(JSON.parse(response.responseText));
+								} catch (e) {
+									this.onerror(response);
+								}
+							}
+						});
+					});
+				},
+
+				like: function(post) {
```

Comment:
unused? probably shouldn't be part of this PR

Reply:
all these functions are the entire point of the PR - i only touched OCP as an *example* of how nice it'll be to have them ready to put into extensions

Reply:
I get that, but they're still dead code. There's no guarantee they'll get used in the future, and they're very likely to break without anyone using them.

Reply:
Servant and Mass Deleter could both use these functions, and I'd rather have this merged first and be able to open PRs for both those extensions rather than choose which to do first - or I could work on them *here* and make this PR even bigger

Reply:
not sure I understand. whats the problem with, once this is merged, opening a new PR that adds one function to xkit_patches and refactors servent to use it?

Reply:
the problem being that if I did that, I would have to either wait for that new PR to be merged before I used them in Mass Deleter, add the same functions in two separate PRs, or refactor multiple unrelated extensions in the same PR

i also dont see the point of withholding them from the codebase when they work and are unlikely to break before theyre used officially, but thats just me